app/src/main/java/io/github/gsantner/memetastic/ui/MemeItemAdapter.java

```diff
@@ -138,14 +174,63 @@ public int getItemCount() {
     }
 
     @Override
-    public void onImageLoaded(Bitmap bitmap, ViewHolder holder) {
+    public void onImageLoaded(Bitmap bitmap, final ViewHolder holder) {
         MemeData.Image dataImage = (MemeData.Image) holder.imageView.getTag();
         Animation animation = AnimationUtils.loadAnimation(_activity, R.anim.fadeinfast);
         holder.imageView.startAnimation(animation);
         if (dataImage.isTemplate) {
             holder.imageButtonFav.startAnimation(animation);
             holder.imageButtonFav.setVisibility(View.VISIBLE);
         }
+
+        if (_app.settings.isHidden(dataImage.fullPath.getAbsolutePath())){
+            holder.imageButtonFav.setVisibility(View.INVISIBLE);
+            holder.imageView.setOnClickListener(null);
+            View longClickView;
+            switch (_itemViewType) {
+                case VIEW_TYPE__ROWS_WITH_TITLE: {
+                    longClickView = holder.itemView;
+                    break;
+                }
+
+                case VIEW_TYPE__PICTURE_GRID:
+                default: {
+                    longClickView =  holder.imageView;
+                    break;
+                }
+            }
+
+            longClickView.setOnLongClickListener(new View.OnLongClickListener() {
+                @Override
```

Comment:
mostly duplicated code from other listener. maybe move the duplication part to a method

Reply:
Okay